Add missing property examples to XDocs for Checkstyle Checks

Checkstyle validates that all configurable properties of checks are demonstrated in the XDocs examples. However, some properties are currently missing from the examples and are temporarily suppressed in the test `XdocsExampleFileTest.java.`

This issue aims to allow new contributors to improve test coverage by adding valid <property> entries to the appropriate Xdocs example files (found under `src/xdocs-examples/resources-noncompilable/... or src/xdocs-examples/resources/...`) and removing the corresponding entry from the suppression list once the example is added.

- Add valid Xdoc example usages (i.e., in /*xml blocks) for the listed properties.
- Ensure that the test `XdocsExampleFileTest` passes without relying on suppression for the properties once they are added.
- Clean up the suppression list by removing each property from the `SUPPRESSED_PROPERTIES_BY_CHECK` map when it’s properly covered.

- [ ] JavadocMetadataScraper: writeXmlOutput
- [ ] IllegalTokenTextCheck: message
- [ ] FinalLocalVariableCheck: validateUnnamedVariables
- [ ] ClassMemberImpliedModifierCheck: violateImpliedStaticOnNestedRecord
- [ ] ClassMemberImpliedModifierCheck: violateImpliedStaticOnNestedEnum
- [ ] ClassMemberImpliedModifierCheck: violateImpliedStaticOnNestedInterface
- [ ] MethodNameCheck: applyToPackage
- [ ] MethodNameCheck: applyToPrivate
- [ ] IllegalImportCheck: regexp
- [ ] IllegalImportCheck: illegalPkgs
- [ ] IllegalImportCheck: illegalClasses
- [ ] IndentationCheck: throwsIndent
- [ ] IndentationCheck: arrayInitIndent
- [ ] IndentationCheck: basicOffset
- [ ] IndentationCheck: braceAdjustment
- [ ] IndentationCheck: lineWrappingIndentation
- [ ] IllegalTypeCheck: legalAbstractClassNames
- [ ] WhitespaceAroundCheck: allowEmptySwitchBlockStatements
- [ ] ConstantNameCheck: applyToPackage
- [ ] ConstantNameCheck: applyToPrivate
- [ ] TypeNameCheck: applyToPackage
- [ ] TypeNameCheck: applyToPublic
- [ ] MissingJavadocTypeCheck: skipAnnotations
- [ ] SummaryJavadocCheck: checkEmptyJavadoc
- [ ] SummaryJavadocCheck: endOfSentenceFormat
- [ ] InterfaceMemberImpliedModifierCheck: violateImpliedFinalField
- [ ] InterfaceMemberImpliedModifierCheck: violateImpliedAbstractMethod
- [ ] InterfaceMemberImpliedModifierCheck: violateImpliedStaticField
- [ ] InterfaceMemberImpliedModifierCheck: violateImpliedPublicMethod
- [ ] InterfaceMemberImpliedModifierCheck: violateImpliedPublicField
- [ ] SuppressWarningsHolder: aliasList
- [ ] JavaNCSSCheck: recordMaximum
- [ ] FinalParametersCheck: ignoreUnnamedParameters
- [ ] MissingJavadocMethodCheck: minLineCount
- [ ] JavadocStyleCheck: checkEmptyJavadoc
- [ ] JavadocStyleCheck: endOfSentenceFormat
- [ ] DescendantTokenCheck: minimumMessage
- [ ] UnusedLocalVariableCheck: allowUnnamedVariables
- [ ] MethodCountCheck: maxPrivate
- [ ] MethodCountCheck: maxPackage
- [ ] MethodCountCheck: maxProtected
- [ ] TrailingCommentCheck: legalComment

> [@smita1078](https://github.com/smita1078)
> 
> ```
> [ERROR]   XdocsExampleFileTest.testAllCheckPropertiesAreUsedInXdocsExamples:122 Xdocs are missing properties:
> Missing property in xdoc: 'skipAnnotations' of MissingJavadocTypeCheck                                                                                                              
> [INFO] 
> [ERROR] Tests run: 5676, Failures: 1, Errors: 0, Skipped: 21
> ```
> 
> Why am I getting this error when removed skipAnnotations from XdocsExampleFileTest.java? The skipAnnotations property is already specified in the configuration file `missingjavadoctype.xml`
> 
> ```
>               <td><a id="skipAnnotations"/><a href="#skipAnnotations">skipAnnotations</a></td>
>               <td>Specify annotations that allow missed documentation. If annotation is present in target sources in multiple forms of qualified name, all forms should be listed in this property.</td>
>               <td><a href="../../property_types.html#String.5B.5D">String[]</a></td>
>               <td><code>Generated</code></td>
>               <td>8.20</td>
> ```

This issue aims to allow new contributors to improve test coverage by adding valid entries to the appropriate Xdocs example files (found under src/xdocs-examples/resources-noncompilable/... or src/xdocs-examples/resources/...) and removing the corresponding entry from the suppression list once the example is added.

Example files are not there, which demonstrate use of `skipAnnotations `property.  Check the folder which has example for missingjavadoctype it doesnot contain one thats why error is there

--- a/config/checkstyle-examples-suppressions.xml
+++ b/config/checkstyle-examples-suppressions.xml
@@ -127,7 +127,7 @@
 
   <!-- violation required by to show behavior of Check -->
   <suppress checks="Indentation"
-    files="indentation[\\/]Example[1-5].java"/>
+    files="indentation[\\/]Example[1-9].java"/>
 
   <!-- To match the behavior with Google style guide -->
   <suppress checks="Indentation"

--- a/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExampleFileTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsExampleFileTest.java
@@ -55,19 +55,6 @@ public class XdocsExampleFileTest {
         "violateExecutionOnNonTightHtml"
     );
 
-    // This list is temporarily suppressed.
-    // Until: https://github.com/checkstyle/checkstyle/issues/17449
-    private static final Map<String, Set<String>> SUPPRESSED_PROPERTIES_BY_CHECK = Map.ofEntries(
-            Map.entry("SuppressWarningsHolder", Set.of("aliasList")),
-            Map.entry("IndentationCheck", Set.of(
-                    "basicOffset",
-                    "lineWrappingIndentation",
-                    "throwsIndent",
-                    "arrayInitIndent",
-                    "braceAdjustment"
-            ))
-    );
-
     @Test
     public void testAllCheckPropertiesAreUsedInXdocsExamples() throws Exception {
         final Map<String, Set<String>> usedPropertiesByCheck =
@@ -87,13 +74,8 @@ public class XdocsExampleFileTest {
             final Set<String> usedProperties =
                 usedPropertiesByCheck.getOrDefault(checkSimpleName, Collections.emptySet());
 
-            final Set<String> suppressedProps =
-                SUPPRESSED_PROPERTIES_BY_CHECK.getOrDefault(
-                    checkSimpleName, Collections.emptySet());
-
             for (String property : definedProperties) {
-                if (!usedProperties.contains(property)
-                        && !suppressedProps.contains(property)) {
+                if (!usedProperties.contains(property)) {
                     failures.add("Missing property in xdoc: '"
                             + property + "' of " + checkSimpleName);
                 }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java
@@ -222,9 +222,20 @@ public final class XdocUtil {
                     parseConfigBlock(xmlBlockMatcher.group(1));
 
                 if (entry != null) {
+                    final String checkClassKey = entry.getKey();
+                    final Set<String> props = entry.getValue();
                     checkToProperties
-                        .computeIfAbsent(entry.getKey(), key -> new HashSet<>())
-                        .addAll(entry.getValue());
+                        .computeIfAbsent(checkClassKey, key -> new HashSet<>())
+                        .addAll(props);
+                    // Also store under module name without "Check" suffix to handle checks
+                    // like SuppressWarningsHolder whose class name does not end with "Check"
+                    if (checkClassKey.endsWith("Check")) {
+                        final String moduleNameKey = checkClassKey.substring(
+                                0, checkClassKey.length() - "Check".length());
+                        checkToProperties
+                            .computeIfAbsent(moduleNameKey, key -> new HashSet<>())
+                            .addAll(props);
+                    }
                 }
             }
         }
@@ -234,30 +245,40 @@ public final class XdocUtil {
     }
 
     private static Map.Entry<String, Set<String>> parseConfigBlock(String configBlock) {
-        final Matcher moduleMatcher =
-                Pattern.compile("<module name=\"([^\"]+)\"").matcher(configBlock);
-        String lastModule = null;
-        while (moduleMatcher.find()) {
-            lastModule = moduleMatcher.group(1);
-        }
-
         final Matcher propMatcher =
                 Pattern.compile("<property name=\"([^\"]+)\"").matcher(configBlock);
         final Set<String> props = new HashSet<>();
+        int firstPropStart = -1;
         while (propMatcher.find()) {
+            if (firstPropStart == -1) {
+                firstPropStart = propMatcher.start();
+            }
             props.add(propMatcher.group(1));
         }
 
         Map.Entry<String, Set<String>> result = null;
-        if (lastModule != null && !props.isEmpty()) {
-            final String checkClassName;
-            if (lastModule.endsWith("Check")) {
-                checkClassName = lastModule;
+        if (!props.isEmpty()) {
+            // Find the last module before the first property to correctly associate
+            // properties with their owning module (handles cases where sibling modules
+            // appear after the module that owns the properties)
+            final Matcher moduleMatcher =
+                    Pattern.compile("<module name=\"([^\"]+)\"")
+                            .matcher(configBlock.substring(0, firstPropStart));
+            String lastModule = null;
+            while (moduleMatcher.find()) {
+                lastModule = moduleMatcher.group(1);
             }
-            else {
-                checkClassName = lastModule + "Check";
+
+            if (lastModule != null) {
+                final String checkClassName;
+                if (lastModule.endsWith("Check")) {
+                    checkClassName = lastModule;
+                }
+                else {
+                    checkClassName = lastModule + "Check";
+                }
+                result = Map.entry(checkClassName, props);
             }
-            result = Map.entry(checkClassName, props);
         }
 
         return result;

--- a/src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckExamplesTest.java
+++ b/src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckExamplesTest.java
@@ -61,4 +61,34 @@ public class IndentationCheckExamplesTest extends AbstractExamplesModuleTestSupp
         final String[] expected = {};
         verifyWithInlineConfigParser(getPath("Example4.java"), expected);
     }
+
+    @Test
+    public void testExample5() throws Exception {
+        final String[] expected = {};
+        verifyWithInlineConfigParser(getPath("Example5.java"), expected);
+    }
+
+    @Test
+    public void testExample6() throws Exception {
+        final String[] expected = {};
+        verifyWithInlineConfigParser(getPath("Example6.java"), expected);
+    }
+
+    @Test
+    public void testExample7() throws Exception {
+        final String[] expected = {};
+        verifyWithInlineConfigParser(getPath("Example7.java"), expected);
+    }
+
+    @Test
+    public void testExample8() throws Exception {
+        final String[] expected = {};
+        verifyWithInlineConfigParser(getPath("Example8.java"), expected);
+    }
+
+    @Test
+    public void testExample9() throws Exception {
+        final String[] expected = {};
+        verifyWithInlineConfigParser(getPath("Example9.java"), expected);
+    }
 }

--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example5.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example5.java
@@ -1,0 +1,56 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="Indentation">
+      <property name="basicOffset" value="2"/>
+    </module>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
+
+// xdoc section -- start
+class Example5 {
+  String field = "example";                  // basicOffset
+  int[] values = {                           // basicOffset
+      10,
+      20,
+      30
+  };
+
+  void processValues() throws Exception {
+    handleValue("Test String", 42);          // basicOffset
+  }
+
+  void handleValue(String aFooString,
+                   int aFooInt) {
+
+    boolean cond1,cond2,cond3,cond4,cond5,cond6;
+    cond1=cond2=cond3=cond4=cond5=cond6=false;
+
+    if (cond1
+        || cond2) {
+      field = field.toUpperCase()
+          .concat(" TASK");
+    }
+
+    if ((cond1 && cond2)
+            || (cond3 && cond4)          // ok, lineWrappingIndentation
+            || !(cond5 && cond6)) {      // ok, lineWrappingIndentation
+      field.toUpperCase()
+           .concat(" TASK")             // ok, lineWrappingIndentation
+           .chars().forEach(c -> {      // ok, lineWrappingIndentation
+             System.out.println((char) c);
+           });
+    }
+  }
+
+  void demonstrateSwitch() throws Exception {
+    switch (field) {
+        case "EXAMPLE": processValues();                        // caseIndent
+        case "COMPLETED": handleValue("Completed Case", 456);   // caseIndent
+    }
+  }
+}
+// xdoc section -- end

--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example6.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example6.java
@@ -1,0 +1,56 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="Indentation">
+      <property name="lineWrappingIndentation" value="8"/>
+    </module>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
+
+// xdoc section -- start
+class Example6 {
+    String field = "example";                  // basicOffset
+    int[] values = {                           // basicOffset
+        10,
+        20,
+        30
+    };
+
+    void processValues() throws Exception {
+        handleValue("Test String", 42);          // basicOffset
+    }
+
+    void handleValue(String aFooString,
+                     int aFooInt) {             // ok, lineWrappingIndentation
+
+        boolean cond1,cond2,cond3,cond4,cond5,cond6;
+        cond1=cond2=cond3=cond4=cond5=cond6=false;
+
+        if (cond1
+                || cond2) {                     // ok, lineWrappingIndentation
+            field = field.toUpperCase()
+                    .concat(" TASK");           // ok, lineWrappingIndentation
+        }
+
+        if ((cond1 && cond2)
+                || (cond3 && cond4)             // ok, lineWrappingIndentation
+                || !(cond5 && cond6)) {         // ok, lineWrappingIndentation
+            field.toUpperCase()
+                    .concat(" TASK")            // ok, lineWrappingIndentation
+                    .chars().forEach(c -> {     // ok, lineWrappingIndentation
+                        System.out.println((char) c);
+                    });
+        }
+    }
+
+    void demonstrateSwitch() throws Exception {
+        switch (field) {
+            case "EXAMPLE": processValues();                        // caseIndent
+            case "COMPLETED": handleValue("Completed Case", 456);   // caseIndent
+        }
+    }
+}
+// xdoc section -- end

--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example7.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example7.java
@@ -1,0 +1,58 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="Indentation">
+      <property name="throwsIndent" value="8"/>
+    </module>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
+
+// xdoc section -- start
+class Example7 {
+    String field = "example";                  // basicOffset
+    int[] values = {                           // basicOffset
+        10,
+        20,
+        30
+    };
+
+    void processValues()
+            throws Exception {                 // ok, throwsIndent
+        handleValue("Test String", 42);          // basicOffset
+    }
+
+    void handleValue(String aFooString,
+                     int aFooInt) {             // indent:8 ; expected: > 4;
+
+        boolean cond1,cond2,cond3,cond4,cond5,cond6;
+        cond1=cond2=cond3=cond4=cond5=cond6=false;
+
+        if (cond1
+            || cond2) {
+            field = field.toUpperCase()
+                .concat(" TASK");
+        }
+
+        if ((cond1 && cond2)
+                || (cond3 && cond4)          // ok, lineWrappingIndentation
+                || !(cond5 && cond6)) {      // ok, lineWrappingIndentation
+            field.toUpperCase()
+                 .concat(" TASK")             // ok, lineWrappingIndentation
+                 .chars().forEach(c -> {      // ok, lineWrappingIndentation
+                     System.out.println((char) c);
+                 });
+        }
+    }
+
+    void demonstrateSwitch()
+            throws Exception {               // ok, throwsIndent
+        switch (field) {
+            case "EXAMPLE": processValues();                        // caseIndent
+            case "COMPLETED": handleValue("Completed Case", 456);   // caseIndent
+        }
+    }
+}
+// xdoc section -- end

--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example8.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example8.java
@@ -1,0 +1,56 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="Indentation">
+      <property name="arrayInitIndent" value="2"/>
+    </module>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
+
+// xdoc section -- start
+class Example8 {
+    String field = "example";                  // basicOffset
+    int[] values = {                           // basicOffset
+      10,                                      // ok, arrayInitIndent
+      20,
+      30
+    };
+
+    void processValues() throws Exception {
+        handleValue("Test String", 42);          // basicOffset
+    }
+
+    void handleValue(String aFooString,
+                     int aFooInt) {             // indent:8 ; expected: > 4;
+
+        boolean cond1,cond2,cond3,cond4,cond5,cond6;
+        cond1=cond2=cond3=cond4=cond5=cond6=false;
+
+        if (cond1
+            || cond2) {
+            field = field.toUpperCase()
+                .concat(" TASK");
+        }
+
+        if ((cond1 && cond2)
+                || (cond3 && cond4)          // ok, lineWrappingIndentation
+                || !(cond5 && cond6)) {      // ok, lineWrappingIndentation
+            field.toUpperCase()
+                 .concat(" TASK")             // ok, lineWrappingIndentation
+                 .chars().forEach(c -> {      // ok, lineWrappingIndentation
+                     System.out.println((char) c);
+                 });
+        }
+    }
+
+    void demonstrateSwitch() throws Exception {
+        switch (field) {
+            case "EXAMPLE": processValues();                        // caseIndent
+            case "COMPLETED": handleValue("Completed Case", 456);   // caseIndent
+        }
+    }
+}
+// xdoc section -- end

--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example9.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example9.java
@@ -1,0 +1,56 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="Indentation">
+      <property name="braceAdjustment" value="2"/>
+    </module>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
+
+// xdoc section -- start
+class Example9 {
+    String field = "example";                  // basicOffset
+    int[] values = {                           // basicOffset
+        10,
+        20,
+        30
+    };
+
+    void processValues() throws Exception {
+        handleValue("Test String", 42);          // basicOffset
+    }
+
+    void handleValue(String aFooString,
+                     int aFooInt) {             // indent:8 ; expected: > 4;
+
+        boolean cond1,cond2,cond3,cond4,cond5,cond6;
+        cond1=cond2=cond3=cond4=cond5=cond6=false;
+
+        if (cond1
+            || cond2) {
+            field = field.toUpperCase()
+                .concat(" TASK");
+        }
+
+        if ((cond1 && cond2)
+                || (cond3 && cond4)          // ok, lineWrappingIndentation
+                || !(cond5 && cond6)) {      // ok, lineWrappingIndentation
+            field.toUpperCase()
+                 .concat(" TASK")             // ok, lineWrappingIndentation
+                 .chars().forEach(c -> {      // ok, lineWrappingIndentation
+                     System.out.println((char) c);
+                 });
+        }
+    }
+
+    void demonstrateSwitch() throws Exception {
+        switch (field) {
+            case "EXAMPLE": processValues();                        // caseIndent
+            case "COMPLETED": handleValue("Completed Case", 456);   // caseIndent
+        }
+    }
+}
+// xdoc section -- end

